### PR TITLE
feat(docs): create documentation site

### PR DESCRIPTION
**Pull request by HaakonSvane**

### CODE REVIEW

File: apps/docs/docusaurus.config.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                    ],
+                },
+                {
+                    type: "docSidebar",
+                    sidebarId: "tutorialSidebar",
+                    position: "left",
+                    label: "Docusaurus tutorial",
+                },
+                {
+                    href: "https://github.com/facebook/docusaurus",
```
Comment (tormodAase) [inline apps/docs/docusaurus.config.ts:85]: Github link in the top right corner should probably link to [https://github.com/equinor/mad](https://github.com/equinor/mad)

File: apps/docs/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+```
+$ USE_SSH=true yarn deploy
+```
+
+Not using SSH:
+
+```
+$ GIT_USER=<Your GitHub username> yarn deploy
+```
```
Comment (tormodAase) [inline apps/docs/README.md:39]: The `yarn` stuff probably doesn't belong here. Are we planning to host this with GitHub pages?

File: apps/docs/static/img/docusaurus-social-card.jpg
Comment (tormodAase) [inline apps/docs/static/img/docusaurus-social-card.jpg:1]: I don't see this image being used anywhere. Where do we find social cards?

File: apps/docs/static/.nojekyll
Comment (tormodAase) [inline apps/docs/static/.nojekyll:1]: What's this?

File: apps/docs/static/img/undraw_docusaurus_mountain.svg
Comment (tormodAase) [inline apps/docs/static/img/undraw_docusaurus_mountain.svg:1]: This image is not being used anywhere

File: apps/docs/static/img/undraw_docusaurus_react.svg
Comment (tormodAase) [inline apps/docs/static/img/undraw_docusaurus_react.svg:1]: Not in use

File: apps/docs/static/img/undraw_docusaurus_tree.svg
Comment (tormodAase) [inline apps/docs/static/img/undraw_docusaurus_tree.svg:1]: Not in use

**Review by tormodAase — CHANGES_REQUESTED** [review]
Awesome! 🔥 🔥 🔥

**Review by HavardNot — COMMENTED** [review]
I guess this will be mostly maintained by developers? What was the reason you went with docusaurus instead of MKDocs? Any issues or shortcomings?

File: apps/docs/docusaurus.config.ts (hunk 1 above)
Comment (HaakonSvane) [reply apps/docs/docusaurus.config.ts:85]: Fixed

File: apps/docs/static/img/undraw_docusaurus_mountain.svg
Comment (HaakonSvane) [reply apps/docs/static/img/undraw_docusaurus_mountain.svg:1]: Removed

File: apps/docs/static/img/undraw_docusaurus_tree.svg
Comment (HaakonSvane) [reply apps/docs/static/img/undraw_docusaurus_tree.svg:1]: Removed

File: apps/docs/README.md (hunk 2 above)
Comment (HaakonSvane) [reply apps/docs/README.md:39]: Yes, that was the plan. Still might have to diddle a bit with the config to get it hosted correctly. Will look into it

Comment (HaakonSvane) [reply apps/docs/README.md:39]: Also removed the yarn stuff and added some more related commands for us :)

File: apps/docs/static/img/docusaurus-social-card.jpg
Comment (HaakonSvane) [reply apps/docs/static/img/docusaurus-social-card.jpg:1]: I updated it with our own image instead. It is configured in the docusaurus config. I think it is displayed as a preview when we link to the docs in messaging apps like Slack

File: apps/docs/static/.nojekyll
Comment (HaakonSvane) [reply apps/docs/static/.nojekyll:1]: Not too sure. I think it is a flag-file that tells the jekyll engine (used by docusaurus??) to ignore the folder during build. It came up after the initialization of docusaurus, so I'm afraid to delete it :O

**Comment by HaakonSvane** [comment]
@HavardNot Docusaurus is built on a js-stack so it integrates nicely with our monorepo and toolings. It also comes with MDX-support straight outta the box, so it is very easy to connect it with our js-docstrings that we have made in the component library (the table is generated automatically from the docstrings we write in mad components 🪄 ) : 
image

Yes, this site will mainly be for and by developers, so I thought it might be easier for them to stick to a JS stack for all things monorepo

**Review by tormodAase — APPROVED** [review]

**Comment by tormodAase** [comment]
@HaakonSvane @HavardNot I agree, having documentation for our tooling separate from other team related documentation makes sense

**Pull request merged**